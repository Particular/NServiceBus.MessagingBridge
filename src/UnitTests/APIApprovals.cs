using NUnit.Framework;
using Particular.Approvals;
using PublicApiGenerator;

[TestFixture]
public class APIApprovals
{
    [Test]
    public void PublicApi()
    {
        var publicApi = typeof(RouterConfiguration).Assembly.GeneratePublicApi(new ApiGeneratorOptions
        {
            ExcludeAttributes = new[] { "System.Runtime.Versioning.TargetFrameworkAttribute", "System.Reflection.AssemblyMetadataAttribute" }
        });

        Approver.Verify(publicApi);
    }
}