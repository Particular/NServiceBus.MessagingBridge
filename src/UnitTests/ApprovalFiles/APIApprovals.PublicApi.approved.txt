[assembly: System.Runtime.CompilerServices.InternalsVisibleTo(@"AcceptanceTesting, PublicKey=00240000048000009400000006020000002400005253413100040000010001007f16e21368ff041183fab592d9e8ed37e7be355e93323147a1d29983d6e591b04282e4da0c9e18bd901e112c0033925eb7d7872c2f1706655891c5c9d57297994f707d16ee9a8f40d978f064ee1ffc73c0db3f4712691b23bf596f75130f4ec978cf78757ec034625a5f27e6bb50c618931ea49f6f628fd74271c32959efb1c5")]
[assembly: System.Runtime.CompilerServices.InternalsVisibleTo(@"UnitTests, PublicKey=00240000048000009400000006020000002400005253413100040000010001007f16e21368ff041183fab592d9e8ed37e7be355e93323147a1d29983d6e591b04282e4da0c9e18bd901e112c0033925eb7d7872c2f1706655891c5c9d57297994f707d16ee9a8f40d978f064ee1ffc73c0db3f4712691b23bf596f75130f4ec978cf78757ec034625a5f27e6bb50c618931ea49f6f628fd74271c32959efb1c5")]
namespace NServiceBus
{
    public class BridgeConfiguration
    {
        public BridgeConfiguration() { }
        public void AddTransport(NServiceBus.BridgeTransport transportConfiguration) { }
        public void DoNotEnforceBestPractices() { }
        public void RunInReceiveOnlyTransactionMode() { }
    }
    public class BridgeEndpoint
    {
        public BridgeEndpoint(string name) { }
        public BridgeEndpoint(string name, string queueAddress) { }
        public void RegisterPublisher(string eventTypeAssemblyQualifiedName, string publisher) { }
        public void RegisterPublisher(System.Type eventType, string publisher) { }
        public void RegisterPublisher<T>(string publisher) { }
    }
    public class BridgeTransport
    {
        public BridgeTransport(NServiceBus.Transport.TransportDefinition transportDefinition) { }
        public bool AutoCreateQueues { get; set; }
        public int Concurrency { get; set; }
        public string ErrorQueue { get; set; }
        public string Name { get; set; }
        public void HasEndpoint(NServiceBus.BridgeEndpoint endpoint) { }
        public void HasEndpoint(string endpointName) { }
        public void HasEndpoint(string endpointName, string endpointAddress) { }
    }
    public static class HostBuilderExtensions
    {
        public static Microsoft.Extensions.Hosting.IHostBuilder UseNServiceBusBridge(this Microsoft.Extensions.Hosting.IHostBuilder hostBuilder, System.Action<NServiceBus.BridgeConfiguration> bridgeConfigurationAction) { }
        public static Microsoft.Extensions.Hosting.IHostBuilder UseNServiceBusBridge(this Microsoft.Extensions.Hosting.IHostBuilder hostBuilder, System.Action<Microsoft.Extensions.Hosting.HostBuilderContext, NServiceBus.BridgeConfiguration> bridgeConfigurationAction) { }
    }
}